### PR TITLE
Harfbuzz freetype coretext

### PR DESCRIPTION
**Pull request by tallytalwar**
The work here makes alfons use coretext for harfbuzz font creation to get shaped glyph from coretext instead of freetype. We still use freetype face for rendering and font atlas texture lookup purposes.

All font loading happens through the `FontSourceHandle` interface. This updates the previous implementation of `systemFont` which was passing around raw font data. `FontSourceHandle` also maintains if a font is a native apple font, to pass that information to `alfons::InputSource` aptly.

Addresses #1513

### CODE REVIEW

**Comment by tallytalwar** [comment]
This is ready for review now.

File: core/include/tangram/platform.h [hunk 1]
```diff
@@ -14,10 +14,14 @@ using FontSourceLoader = std::function<std::vector<char>()>;
...
 struct FontSourceHandle {
     FontSourceHandle(std::string _path) : path(_path) {}
+    FontSourceHandle(std::string _path, bool _appleFont) : path(_path), appleFont(_appleFont) {}
     FontSourceHandle(FontSourceLoader _loader) : load(_loader) {}
+    FontSourceHandle() {}
 
     std::string path;
-    FontSourceLoader load;
+    FontSourceLoader load = nullptr;
+    bool appleFont = false;
```
Comment (hjanetzek) [inline core/include/tangram/platform.h]: Can we rename this flag to `systemFontName` or `fontName` ?

Comment (tallytalwar) [reply core/include/tangram/platform.h]: updated aptly to reflect the usage.

**Comment by tallytalwar** [comment]
Updated submodules to master commits for alfons and harfbuzz-icu-freetype and addressed the comment @hjanetzek mentioned.
Will merge once approved!

**Comment by tallytalwar** [comment]
Rebased with master, merging when CI goes green.

**Comment by matteblair** [comment]
@hjanetzek You reviewed this, right? Can you give this a ✅ if it looks good?

File: core/include/tangram/platform.h [hunk 2]
```diff
@@ -13,11 +13,15 @@ using UrlCallback = std::function<void(std::vector<char>&&)>;
 using FontSourceLoader = std::function<std::vector<char>()>;
 
 struct FontSourceHandle {
-    FontSourceHandle(std::string _path) : path(_path) {}
+    FontSourceHandle(std::string _path) : pathOrFontName(_path) {}
+    FontSourceHandle(std::string _pathOrFontName, bool _isFontName) : pathOrFontName(_pathOrFontName), isFontName(_isFontName) {}
```
Comment (karimnaaji) [inline core/include/tangram/platform.h]: Leaving this as a note for future cleanup on this: Using `Url` class here would be cleaner overhaul, only adding a dependency for this class from platform code.

Comment (tallytalwar) [reply core/include/tangram/platform.h]: As discussed in slack, makes sense to do this cleanup with `fontConfig` work for linux platforms.

**Pull request merged**